src/nspawn/nspawn.c

```diff
@@ -2594,9 +2594,15 @@ static int inner_child(
                 return -ESRCH;
         }
 
-        r = mount_systemd_cgroup_writable("", arg_unified_cgroup_hierarchy);
-        if (r < 0)
-                return r;
+	if (cg_ns_supported()) {
+		r = unshare(CLONE_NEWCGROUP);
+		if (r < 0)
```

Comment:
> Sorry, I'm a little confused as to what you're getting at here.

@brauner , sorry.
I mean:
> By default, nspawn will use the unified hierarchy for the containers if the host uses the
unified hierarchy, and the legacy hierarchy otherwise.

Your patch doesn't work as expected: https://github.com/systemd/systemd/pull/3589#discussion_r68485933
```
container# grep cgroup /proc/self/mounts
tmpfs /sys/fs/cgroup tmpfs ro,nosuid,nodev,noexec,mode=755 0 0
cgroup /sys/fs/cgroup/systemd cgroup rw,nosuid,nodev,noexec,relatime,xattr,release_agent=/usr/lib/systemd/systemd-cgroups-agent,name=systemd 0 0
cgroup /sys/fs/cgroup/freezer cgroup rw,nosuid,nodev,noexec,relatime,freezer 0 0
cgroup /sys/fs/cgroup/net_cls cgroup rw,nosuid,nodev,noexec,relatime,net_cls 0 0
cgroup /sys/fs/cgroup/cpu,cpuacct cgroup rw,nosuid,nodev,noexec,relatime,cpu,cpuacct 0 0
cgroup /sys/fs/cgroup/devices cgroup rw,nosuid,nodev,noexec,relatime,devices 0 0
cgroup /sys/fs/cgroup/cpuset cgroup rw,nosuid,nodev,noexec,relatime,cpuset 0 0
```
`master` works fine:
```
container# grep cgroup /proc/self/mounts
cgroup /sys/fs/cgroup cgroup2 ro,nosuid,nodev,noexec,relatime 0 0
cgroup /sys/fs/cgroup/machine.slice/machine-nspawn\134x2droot.scope cgroup2 rw,nosuid,nodev,noexec,relatime 0 0
```

Yeah
```sh
systemd-nspawn -D /some/rootfs -b 'systemd.unified_cgroup_hierarchy=1'
```
mounts the `v2`-hierarchy. But we should do this by default (i.e. without `systemd.unified_cgroup_hierarchy=1`)